src/general/NewGameSettings.cs

```diff
@@ -0,0 +1,638 @@
+        if (Visible)
+            return;
+
+        Show();
+    }
+
+    public void ReportValidityOfGameSeed(bool valid)
+    {
+        if (valid)
+        {
+            GUICommon.MarkInputAsValid(gameSeed);
+            GUICommon.MarkInputAsValid(gameSeedAdvanced);
+            confirmButton.Disabled = false;
+            confirmButton.HintTooltip = TranslationServer.Translate("CONFIRM_NEW_GAME_BUTTON_TOOLTIP");
+        }
+        else
+        {
+            GUICommon.MarkInputAsInvalid(gameSeed);
+            GUICommon.MarkInputAsInvalid(gameSeedAdvanced);
+            confirmButton.Disabled = true;
+            confirmButton.HintTooltip = TranslationServer.Translate("CONFIRM_NEW_GAME_BUTTON_TOOLTIP_DISABLED");
+        }
+    }
+
+    private string GenerateNewRandomSeed()
+    {
+        var random = new Random();
+        return random.Next().ToString();
+    }
+
+    private void SetSeed(string text)
+    {
+        isGameSeedValid = int.TryParse(text, out int seed) && seed > 0;
+        ReportValidityOfGameSeed(isGameSeedValid);
+        if (isGameSeedValid)
+            settings.Seed = seed;
+    }
+
+    /// <summary>
+    ///   Changes the active settings tab that is displayed, or returns if the tab is already active.
+    /// </summary>
+    private void ChangeSettingsTab(string newTabName)
+    {
+        // Convert from the string binding to an enum.
+        SelectedOptionsTab selection = (SelectedOptionsTab)Enum.Parse(typeof(SelectedOptionsTab), newTabName);
+
+        // Pressing the same button that's already active, so just return.
+        if (selection == selectedOptionsTab)
+            return;
+
+        difficultyTab.Hide();
+        planetTab.Hide();
+        miscTab.Hide();
+
+        switch (selection)
+        {
+            case SelectedOptionsTab.Difficulty:
+                difficultyTab.Show();
+                difficultyTabButton.Pressed = true;
+                break;
+            case SelectedOptionsTab.Planet:
+                planetTab.Show();
+                planetTabButton.Pressed = true;
+                break;
+            case SelectedOptionsTab.Miscellaneous:
+                miscTab.Show();
+                miscTabButton.Pressed = true;
+                break;
+            default:
+                GD.PrintErr("Invalid tab");
+                break;
+        }
+
+        GUICommon.Instance.PlayButtonPressSound();
+        selectedOptionsTab = selection;
+    }
+
+    /*
+      GUI Control Callbacks
+    */
+
+    private void OnBackPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        Exit();
+    }
+
+    private bool Exit()
+    {
+        EmitSignal(nameof(OnNewGameSettingsClosed));
+        return true;
+    }
+
+    private void OnAdvancedPressed()
+    {
+        ProcessAdvancedSelection(true);
+    }
+
+    private void ProcessAdvancedSelection(bool playSound)
+    {
+        if (playSound)
+            GUICommon.Instance.PlayButtonPressSound();
```

Comment:
The usual non-written way we do it is to only call the play button press sound from the callbacks, so that the actual logic containing methods doesn't have to know whether to play the sound or not.